webapp/src/pages/Settings.tsx

```diff
@@ -159,6 +159,33 @@ const displayReadonlyFields = (label: string, value: string | null) => (
   />
 );
 
+const StringField: React.FC<
+  Omit<TextFieldProps, "onChange"> & {
+    label: string;
+    value: string | null;
+    onChange: (newValue: string) => void;
+  }
+> = ({ label, value, onChange, ...props }) => {
+  return (
```

Comment:
You can remove the braces and the `return` to save one indent (and two lines).
```suggestion
> = ({ label, value, onChange, ...props }) => (
```